yarn-project/simulator/src/public/dual_side_effect_trace.ts

```diff
@@ -177,6 +177,17 @@ export class DualSideEffectTrace implements PublicSideEffectTraceInterface {
    * Convert this trace to a PublicExecutionResult for use externally to the simulator.
    */
   public toPublicExecutionResult(
+    /** How much gas was left after this public execution. */
+    endGasLeft: Gas,
+    /** The call's results */
+    avmCallResults: AvmContractCallResult,
+  ): EnqueuedPublicCallExecutionResultWithSideEffects {
+    return this.enqueuedCallTrace.toPublicExecutionResult(endGasLeft, avmCallResults);
+  }
+  /**
+   * Convert this trace to a PublicExecutionResult for use externally to the simulator.
+   */
+  public toPublicFunctionCallResult(
     /** The execution environment of the nested call. */
     avmEnvironment: AvmExecutionEnvironment,
```

Comment:
toPublicExecutionResult is meant to give you enqueued call results, and toPublicFunctionCallResult is meant to give you the nested call results used by today's AVM circuit